filebeat/module/apache2/access/ingest/default.json

```diff
@@ -4,8 +4,8 @@
     "grok": {
       "field": "message",
       "patterns":[
-        "%{IPORHOST:apache2.access.remote_ip} - %{DATA:apache2.access.user_name} \\[%{HTTPDATE:apache2.access.time}\\] \"%{WORD:apache2.access.method} %{DATA:apache2.access.url} HTTP/%{NUMBER:apache2.access.http_version}\" %{NUMBER:apache2.access.response_code} (?:%{NUMBER:apache2.access.body_sent.bytes}|-)( \"%{DATA:apache2.access.referrer}\")?( \"%{DATA:apache2.access.agent}\")?",
-        "%{IPORHOST:apache2.access.remote_ip} - %{DATA:apache2.access.user_name} \\[%{HTTPDATE:apache2.access.time}\\] \"-\" %{NUMBER:apache2.access.response_code} -"
+        "%{IP:source.ip} - %{DATA:user.name} \\[%{HTTPDATE:apache2.access.time}\\] \"%{WORD:http.request.method} %{DATA:url.original} HTTP/%{NUMBER:http.version}\" %{NUMBER:http.response.status_code} (?:%{NUMBER:http.response.body_sent.bytes}|-)( \"%{DATA:http.request.referer}\")?( \"%{DATA:user_agent.original}\")?",
+        "%{IP:source.ip} - %{DATA:user.name} \\[%{HTTPDATE:apache2.access.time}\\] \"-\" %{NUMBER:http.response.status_code} -"
```

Comment:
You should define a fallback for the host at the beginning of the line. Using the `IP` grok will fail if it's a hostname.

You could define the fallback like was suggested in https://github.com/elastic/ecs/issues/166#issuecomment-435721565. In our case, it could look like:

```
"(?:%{IP:source.ip}|%{NOTSPACE:source.domain}) - %{DATA:user.name} ...
```

Note also that simply attempting to store a hostname at `source.ip` (of type `ip`) will fail to be indexed with a `mapper_parsing_exception`.

Reply:
adjusted it with your proposal.